iothub/device/src/InternalClient.cs

```diff
@@ -1167,18 +1169,14 @@ public async Task<Twin> GetTwinAsync()
         /// For the complete device twin object, use Microsoft.Azure.Devices.RegistryManager.GetTwinAsync(string deviceId).
         /// </summary>
         /// <returns>The device twin object for the current device</returns>
-        public Task<Twin> GetTwinAsync(CancellationToken cancellationToken)
+        public async Task<Twin> GetTwinAsync(CancellationToken cancellationToken)
         {
-            // `GetTwinAsync` shall call `SendTwinGetAsync` on the transport to get the twin state.
-            try
-            {
-                return InnerHandler.SendTwinGetAsync(cancellationToken);
-            }
-            catch (IotHubCommunicationException ex) when (ex.InnerException is OperationCanceledException)
```

Comment:
Previously we returned the Task as-is, without awaiting its completion. For this reason, the catch block was never executed. Now that we're awaiting the Task, any thrown exception would be caught in the catch block and transformed, if relevant. Since this will change the behavior, we'll remove the catch block itself.